test/test_strerror.cpp

```diff
@@ -54,3 +55,90 @@ TEST(test_strerror, get_error) {
     "' did not cause the expected error message.";
 #endif
 }
+
+/*
+   Define the blueprint of a mock identified by `strerror_r_proto`
+   strerror_r possible signatures:
+
+   * int strerror_r(int errnum, char *buf, size_t buflen); (Case 1)
+   * char *strerror_r(int errnum, char *buf, size_t buflen); (Case 2)
+   * errno_t strerror_s( char *buf, rsize_t bufsz, errno_t errnum ); (Case 3)
+*/
+
+#if defined(_WIN32)
+mmk_mock_define(strerror_s_mock, errno_t, char *, rsize_t, errno_t);
+#elif defined(_GNU_SOURCE) && (!defined(ANDROID) || __ANDROID_API__ >= 23)
+mmk_mock_define(strerror_r_mock, char *, int, char *, size_t);
+#else
+mmk_mock_define(strerror_r_mock, int, int, char *, size_t);
+#endif
+
+const char expected_error_msg[] = "Failed to get error";
+#if defined(_WIN32)
+// Function to be called for (Case 3)
+errno_t mocked_windows_strerror(char * buf, rsize_t bufsz, errno_t errnum)
+{
+  (void) errnum;
+  unsigned char index_err = 0;
+
+  while (buf && bufsz--) {
+    buf[index_err] = expected_error_msg[index_err];
+    index_err++;
+  }
+  return errnum;
+}
+#else
+// Function to be called for (Case 1)
+char * mocked_gnu_strerror(int errnum, char * buf, size_t buflen)
+{
+  (void) errnum;
+  const char error_msg[] = "Failed to get error";
+  unsigned char index_err = 0;
+  while (buf && buflen--) {
+    buf[index_err] = error_msg[index_err];
+    index_err++;
+  }
+  return buf;
+}
+#endif
+
+/* Mocking test example */
+TEST(test_strerror, test_mock) {
```

Comment:
@Blast545 I wonder, would it be easier to read if this was split into three separate tests? There's little reuse beyond a few lines.

Reply:
Didn't consider it yesterday, but I like the idea of separating the tests for readability, I'll add separate them.